src/pu-emmc.c

```diff
@@ -164,6 +158,112 @@ emmc_create_partition(PuEmmc *self,
     return TRUE;
 }
 
+static gboolean
+pu_emmc_check_raw_overwrite(PuEmmc *emmc,
+                            GError **error)
+{
+    g_return_val_if_fail(emmc != NULL, FALSE);
+    g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+    if (emmc->partitions && emmc->raw) {
+        PuEmmcPartition *part = emmc->partitions->data;
+        gsize part_start = part->offset * emmc->device->sector_size;
+        gsize raw_start;
+        gsize raw_end;
+
+        for (GList *b = emmc->raw; b != NULL; b = b->next) {
+            PuEmmcBinary *bin = b->data;
+
+            raw_start = bin->output_offset * emmc->device->sector_size;
+            raw_end = bin->input->_size;
+            raw_end += raw_start;
+            raw_end -= bin->input_offset * emmc->device->sector_size;
+
+            if (raw_end > part_start) {
+                g_set_error(error, PU_ERROR, PU_ERROR_FAILED,
+                            "Raw binary overlaps with first partition");
+                return FALSE;
+            }
+
+            /* Check against all binaries after the current one */
+            for (GList *i = b->next; i != NULL; i = i->next) {
+                PuEmmcBinary *bin2 = i->data;
+
+                gsize raw2_start = bin2->output_offset * emmc->device->sector_size;
+                gsize raw2_end;
+                raw2_end = bin2->input->_size;
+                raw2_end += raw2_start;
+                raw2_end -= bin2->input_offset * emmc->device->sector_size;
+
+                if (!(raw_end < raw2_start || raw_start > raw2_end)) {
+                    g_set_error(error, PU_ERROR, PU_ERROR_FAILED,
+                                "Raw binary overlaps with other raw binary");
+                    return FALSE;
+                }
+            }
+        }
+    }
+
+    return TRUE;
+}
+
+static gboolean
+pu_emmc_validate_config(PuFlash *flash,
+                        GError **error)
+{
+    PuEmmc *self = PU_EMMC(flash);
+    gboolean skip_checksums = FALSE;
+
+    g_return_val_if_fail(flash != NULL, FALSE);
+    g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+    g_object_get(flash,
+                 "skip-checksums", &skip_checksums,
+                 NULL);
+
+    if (!skip_checksums)
+        for (GList *i = self->input_files; i != NULL; i = i->next) {
+            PuInput *input = i->data;
+            if (!pu_validate_file(input, error))
+                return FALSE;
+        }
```

Comment:
Use curly braces to enclose the if-statement content. It's currently a little confusing to read.

Reply:
Fixed